### PR TITLE
Feature/workflow updates

### PR DESCRIPTION
**Pull request by reyhanlioglu**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new enum `NeoNavigationType` with associated methods for JSON serialization and deserialization.
  - Added a new class `SignalrTransitionData` encapsulating data related to signalr transitions.
- **Refactor**
  - Updated import statement for `neo_navigation_type.dart` to point to a different location.
  - Refactored log messages to include the class name in square brackets.
  - Modified the `listenForTransitionEvents` method to handle and parse transition data differently.
  - Modified the `onPageNavigation` property to accept a `SignalrTransitionData` object.
- **Documentation**
  - Added a trailing comma after the `NeoErrorMessage.defaultErrorEnglish` declaration within the `defaultErrorMessages` list.
- **Chores**
  - Modified the `NeoSignalRTransitionFromJson` function to accommodate additional property assignments.
  - Removed the import for `bloc/bloc.dart` and added an import for `flutter_bloc/flutter_bloc.dart`.
  - Removed imports for `neo_crashlytics.dart` and `neo_core_widget_registerer.dart` from `neo_core.dart`.
  - Removed a method call to `NeoCoreWidgetRegisterer().init()` within the `init` method of the `NeoCore` class.

### CODE REVIEW

File: lib/core/navigation/models/signalr_transition_data.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  factory SignalrTransitionData.decode(String jsonString) {
+    final Map<String, dynamic> jsonMap = jsonDecode(jsonString);
+    return SignalrTransitionData(
+      navigationPath: jsonMap[_Constant.keyNavigationPath],
+      navigationType: NeoNavigationType.fromJson(jsonMap[_Constant.keyNavigationType]),
+      pageId: jsonMap[_Constant.keyPageId],
+      viewSource: jsonMap[_Constant.keyViewSource],
+      initialData: jsonMap[_Constant.keyInitialData],
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/navigation/models/signalr_transition_data.dart:57]: No need, encode works properly with all required fields

File: lib/core/navigation/models/signalr_transition_data.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/navigation/models/signalr_transition_data.dart:60]: It's a data class and parameter names are obvious

File: lib/core/network/managers/signalr_connection_manager.dart [hunk 3]
```diff
@@ -56,43 +58,69 @@ class SignalrConnectionManager {
...
+          navigationPath: navigationPath,
+          navigationType: NeoNavigationType.fromJson(navigationType.orEmpty),
+          pageId: ongoingTransition?.pageId ?? "",
+          viewSource: ongoingTransition?.viewSource ?? "",
+          initialData: ongoingTransition?.initialData ?? {},
+        ),
+      );
+    } else if ((ongoingTransition?.errorMessage.isNotEmpty ?? false) && onError != null) {
+      onError(ongoingTransition!.errorMessage);
+    }
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/network/managers/signalr_connection_manager.dart]: Done

File: lib/core/network/models/neo_signalr_transition.dart [hunk 4]
```diff
@@ -20,9 +20,18 @@ class NeoSignalRTransition extends Equatable {
...
+  final String viewSource;
+
   @JsonKey(name: "page", defaultValue: {})
   final Map<String, dynamic> pageDetails;
 
+  @JsonKey(name: "data", defaultValue: {})
+  final Map<String, dynamic> initialData;
+
   @JsonKey(name: "additionalData", defaultValue: null)
   final Map<String, dynamic>? additionalData;
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/network/models/neo_signalr_transition.dart:37]: Done

File: lib/core/navigation/models/signalr_transition_data.dart [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+class _Constant {
+  static const keyNavigationPath = "navigationPath";
+  static const keyNavigationType = "navigationType";
+  static const keyPageId = "pageId";
+  static const keyViewSource = "viewSource";
+  static const keyInitialData = "initialData";
+}
+
+class SignalrTransitionData {
```
Comment (yildirayyorulmaz) [inline lib/core/navigation/models/signalr_transition_data.dart:25]: Neden json_serializable kullanmadin, bi sebebi var mi

File: lib/core/network/managers/signalr_connection_manager.dart [hunk 6]
```diff
@@ -56,45 +58,74 @@ class SignalrConnectionManager {
...
+  NeoSignalRTransition? _parseOngoingTransition(List<Object?> transitions, String transitionId) {
+    return transitions
+        .map((transition) {
+          try {
+            return NeoSignalRTransition.fromJson(jsonDecode(transition is String ? transition : "{}"));
+          } catch (_) {
+            return null;
+          }
+        })
+        .where((element) => element != null)
```
Comment (yildirayyorulmaz) [inline lib/core/network/managers/signalr_connection_manager.dart:90]: opt: whereNotNull kullanilabilir

File: lib/core/navigation/models/signalr_transition_data.dart (hunk 5 above)
Comment (reyhanlioglu) [reply lib/core/navigation/models/signalr_transition_data.dart:25]: çünkü json'a çevirmiyorum stringify ediyorum, return type map değil string

File: lib/core/navigation/models/neo_navigation_type.dart [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  @JsonValue('push-as-root')
+  pushAsRoot('push-as-root'),
+
+  @JsonValue('popup')
+  popup('popup'),
+
+  @JsonValue('bottom-sheet')
+  bottomSheet('bottom-sheet');
+
+  final String value;
```
Comment (yildirayyorulmaz) [inline lib/core/navigation/models/neo_navigation_type.dart:35]: opt: private olabilir

File: lib/core/network/managers/signalr_connection_manager.dart (hunk 6 above)
Comment (reyhanlioglu) [reply lib/core/network/managers/signalr_connection_manager.dart]: Done

File: lib/core/navigation/models/neo_navigation_type.dart (hunk 7 above)
Comment (reyhanlioglu) [reply lib/core/navigation/models/neo_navigation_type.dart]: Done

**Review by mcanyigincommencis — APPROVED** [review]

**Review by ayhncvdr2 — APPROVED** [review]

**Review by yildirayyorulmaz — APPROVED** [review]

File: lib/core/navigation/models/neo_navigation_type.dart (hunk 7 above)
Comment (reyhanlioglu) [reply lib/core/navigation/models/neo_navigation_type.dart]: Hatta value komple gereksizmiş sildim

**Pull request merged**